services/user/Permissions/plugin/wit/world.wit

```diff
@@ -1,39 +1,44 @@
 package permissions:plugin;
 
 interface types {
-    record prompt-context {
-        user: string,
-        caller: string,
-        callee: string,
+    /// This represents the trust threshold for a third-party application to
+    /// perform an action on the user's behalf. It's just a hint to help guide the user.
+    record trust {
+        /// Valid values: `0`-`6`
+        /// * `0`: Anyone can call
+        /// * `1`-`5`: Higher number = higher required trust
+        /// * `6`: No one can call (except the callee app itself)
+        level: u8,
+
+        /// A description that will be placed on the UI shown to the user, following the text,
+        /// "By authorizing, you trust <caller> with these abilities:"
+        ///
+        /// The description should naturally follow that preface. For example:
+        /// Good description: "- Bake a cake"
+        /// Bad description: "This will allow the caller to bake a cake."
+        description: string,
```

Comment:
@swatanabe I don't really like this interface. I was trying to encourage consistency so that the frond-end would look like a standard oauth approval across all permissions prompts, but this just seems like a silly api. 

I also don't like that these instructions couple the provided descriptions to the implementation of the permissions prompt front-end.

...I guess it should just be a string that contains whatever the app thinks is relevant for the user to know, and the permissions front-end will make no assumptions about its contents?

Or... lmk if you have any other idea.